src/sentry/static/sentry/app/utils/queryString.tsx

```diff
@@ -37,7 +37,14 @@ export function appendTagCondition(
 ): string {
   let currentQuery = Array.isArray(query) ? query.pop() : isString(query) ? query : '';
 
-  if (isString(value) && value.indexOf(' ') > -1) {
+  // The user key values have additional key data inside them.
+  if (key === 'user' && value.includes(':')) {
```

Comment:
Maybe we could update this method, fixing all the use cases mentioned here https://github.com/getsentry/sentry/pull/17507

Reply:
It is an adjacent but not entirely related problem. The user tag can contain values like `id:1` which cannot be combined with the tag key as they create conditions like `user:id:1` which is an invalid value in all search boxes. The user tag is a ❄️ as it contains structured data.

The `device.name` and `error.mechanism` attributes are not generally converted into tags that contain structured data.